janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -726,6 +768,26 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
         }
     }
 
+    private List<Map<String, Object>> getParameters(KeyInformation.StoreRetriever storeRetriever,
+                                                    List<IndexEntry> entries,
+                                                    Predicate<Cardinality> predicate) {
```

Comment:
(nitpick) I think this `predicate` is a bit confusing. It's not clear from it's name what it should actually test and it seems to be only necessary for the addition case.
Would it make sense to create two different methods for additions and deletions here? If you want to share the code between both, then you could add a third method with the actual implementation, but then I would probably use a boolean flag instead that has a name like `skipSingleCardinality` to get rid of the predicate.

Reply:
What do you think if we had `cardinalityToSkip` vararg where you can enumerate cardinality?